src/main/java/core/basesyntax/strategy/operation/impl/Balance.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.operation.Operation;
+
+public class Balance implements Operation {
+    @Override
+    public void apply(Transaction transaction) {
+        Fruit fruit = transaction.getFruit();
+        if (!Storage.storage.containsKey(fruit)) {
+            Storage.storage.put(fruit, transaction.getSum());
+        } else {
+            Integer currentQuantity = Storage.storage.get(fruit) + transaction.getSum();
+            Storage.storage.put(fruit, currentQuantity);
+        }
+    }
+}
```

Comment:
looks like we have code duplication here and in other implementations of `Operation`
can we move common logic to a separate class like `FruitService`?